swapserverrpc/instantout.proto

```diff
@@ -0,0 +1,128 @@
+syntax = "proto3";
+
+// We can't change this to swapserverrpc, it would be a breaking change because
+// the package name is also contained in the HTTP URIs and old clients would
+// call the wrong endpoints. Luckily with the go_package option we can have
+// different golang and RPC package names to fix protobuf namespace conflicts.
+package looprpc;
+
+option go_package = "github.com/lightninglabs/loop/swapserverrpc";
+
+service InstantSwapServer {
+    // RequestInstantLoopOut initiates an instant loop out swap.
+    rpc RequestInstantLoopOut (InstantLoopOutRequest)
+        returns (InstantLoopOutResponse);
+
+    // PollPaymentAccepted polls the server to see if the payment has been
+    // accepted.
+    rpc PollPaymentAccepted (PollPaymentAcceptedRequest)
+        returns (PollPaymentAcceptedResponse);
+
+    // InitHtlcSig is called by the client to initiate the htlc sig exchange.
+    rpc InitHtlcSig (InitHtlcSigRequest) returns (InitHtlcSigResponse);
+
+    // PushHtlcSig is called by the client to push the htlc sigs to the server.
+    rpc PushHtlcSig (PushHtlcSigRequest) returns (PushHtlcSigResponse);
+
+    // PushPreimage is called by the client to push the preimage to the server.
+    // This returns the musig2 signatures that the client needs to sweep the
+    // htlc.
+    rpc PushPreimage (PushPreimageRequest) returns (PushPreimageResponse);
+}
+
+message InstantLoopOutRequest {
+    // The amount to loop out in satoshis.
+    uint64 amt = 1;
+
+    // Htlc related fields:
+    // The key for the htlc preimage spending path.
+    bytes receiver_key = 2;
+
+    // T hash of the preimage that will be used to settle the htlc.
+    bytes swap_hash = 3;
+
+    // The requested absolute block height of the on-chain htlc. This is
+    // subjected to min and max constraints as reported in the LoopOutTerms
```

Comment:
is this still matching the terms constraints after our sane default value adjustment?

Reply:
Is the doc up-to-date?